Lambdas/GHHooks/EventHandler/Handlers/ReleaseHandler.swift

```diff
@@ -0,0 +1,414 @@
+                tag_name: newVersion,
+                target_commitish: pr.base.ref,
+                name: "\(newVersion) - \(pr.title)",
+                body: body,
+                draft: false,
+                prerelease: isPrerelease,
+                make_latest: isPrerelease ? ._false : ._true
+            ))
+        ))
+
+        switch response {
+        case let .created(created):
+            switch created.body {
+            case let .json(release):
+                return release
+            }
+        default: break
+        }
+
+        throw Errors.httpRequestFailed(response: response)
+    }
+
+    func sendComment(release: Release) async throws {
+        /// `"Issues" create comment`, but works for PRs too. Didn't find an endpoint for PRs.
+        let response = try await context.githubClient.issues_create_comment(.init(
+            path: .init(
+                owner: repo.owner.login,
+                repo: repo.name,
+                issue_number: number
+            ),
+            body: .json(.init(
+                body: """
+                These changes are now available in [\(release.tag_name)](\(release.html_url))
+                """
+            ))
+        ))
+
+        switch response {
+        case .created: return
+        default:
+            throw Errors.httpRequestFailed(response: response)
+        }
+    }
+
+    /**
+     - A user who appears in a given repo's code owners file should NOT be credited as either an author or reviewer for a release in that repo (but can still be credited for releasing it).
+     - The user who authored the PR should be credited unless they are a code owner. Such a credit should be prominent and - as the GitHub changelog generator does - include a notation if it's that user's first merged PR.
+     - Any users who reviewed the PR (even if they requested changes or did a comments-only review without approving) should also be credited unless they are code owners. Such a credit should be less prominent than the author credit, something like a "thanks to ... for helping to review this release"
+     - The release author (user who merged the PR) should always be credited in a release, even if they're a code owner. This credit should be the least prominent, maybe even just a footnote (since it will pretty much always be a owner/maintainer).
+     */
+    func makeReleaseBody(
+        mergedBy: NullableUser,
+        previousVersion: String,
+        newVersion: String
+    ) async throws -> String {
+        let codeOwners = try await getCodeOwners()
+        let contributors = try await getExistingContributorIDs()
+        let isNewContributor = isNewContributor(
+            codeOwners: codeOwners,
+            existingContributors: contributors
+        )
+        let reviewers = try await getReviewersToCredit(codeOwners: codeOwners)
+        let isCodeOwner = codeOwners.contains(pr.user.login) ||
+        (pr.user.name.map { codeOwners.contains($0) } ?? false)
```

Comment:
Okay, I've now seen this particular OR expression repeated in enough places to say you should add something along these lines:
```swift
struct CodeOwners {
    let names: Set<String>
    
    static func get(for fullRepoName: String, httpClient: HTTPClient, logger: Logger) async throws -> CodeOwners {
        // the appropriately modified content of `getCodeOwners()`
        return .init(names: parsed)
    }

    func contains(user: Components.Schemas.simple_user) -> Bool {
        !self.names.intersection([user.login, user.name].compactMap{$0}).isEmpty
    }
}
```

Reply:
considering this is just in a file, i can just do `private extension Set<String>` i guess?